core-backup/src/test/scala/io/aiven/guardian/kafka/backup/MockedBackupClientInterface.scala

```diff
@@ -84,10 +87,23 @@ class MockedBackupClientInterface(override val kafkaClientInterface: MockedKafka
     */
   override type BackupResult = Done
 
-  override type State = Nothing
+  override type State = Unit
 
-  override def getCurrentUploadState(key: String): Future[UploadStateResult] =
-    Future.successful(UploadStateResult.empty)
+  override def getCurrentUploadState(key: String): Future[UploadStateResult] = {
+    val keys  = backedUpData.asScala.map { case (k, _) => k }.toVector.distinct.sortBy(Utils.keyToOffsetDateTime)
+    val index = keys.indexOf(key)
+
+    val uploadStateResult =
+      if (index < 0)
+        UploadStateResult.empty
+      else {
+        val previous = keys.lift(index - 1).map(k => PreviousState((), k))
```

Comment:
The `lift` method is the exact same as getting an element out of a collection by index with the difference being it returns an `Option` rather than throwing an exception of the element doesn't exist.